index.bs

```diff
@@ -308,7 +309,8 @@ resources are no longer accessible from JavaScript.
 <pre class="idl">
 interface MediaStreamTrackGenerator : MediaStreamTrack {
     constructor(MediaStreamTrackGeneratorInit init);
-    attribute WritableStream writable;  // VideoFrame or AudioFrame
+    attribute WritableStream writable;
+    undefined setMuted(boolean newState);
```

Comment:
I guess you don't want to make writable heterogeneous? (I.e. allow signals as well as frames?)

(I certainly wouldn't want that for MSTP.readable -- it would make the code very complicated.)

If the idea is that writable would be written on a Worker, wouldn't we need to be able to call this from the same Worker?

Perhaps we could let muted reflect writable.locked? I.e., the track starts off muted, when you call writable.getWriter(), it becomes unmuted, and if you call writer.releaseLock() it becomes muted again? (There might be an issue with synchronization here, so probably best to state it's asynchronously or loosely tied to the locked state.)

Reply:
> I guess you don't want to make writable heterogeneous? (I.e. allow signals as well as frames?)
> 
Is that about the comment removal? That's a drive-by for consistency with MSTP (alternatively I can add the comment to MSTP.readable).
There are no signals anymore anyway.

> (I certainly wouldn't want that for MSTP.readable -- it would make the code very complicated.)
Nothing is changing in that regard.

> 
> If the idea is that writable would be written on a Worker, wouldn't we need to be able to call this from the same Worker?
>  

It would be  good, but I don't see it as a hard requirement since the muted state for tracks or now is only visible on the Window scope.
Once tracks are transferrable, we can also make MSTG transferrable. and it will be available there.

> Perhaps we could let muted reflect writable.locked? I.e., the track starts off muted, when you call writable.getWriter(), it becomes unmuted, and if you call writer.releaseLock() it becomes muted again? (There might be an issue with synchronization here, so probably best to state it's asynchronously or loosely tied to the locked state.)

That looks like  a  very interesting idea.. I have to check if it's viable to spec it  that way  in tterms of spec algorithms.